tests/foreman/rhai/test_rhai_client.py

```diff
@@ -14,70 +14,73 @@
-        with manifests.clone() as manifest:
-            upload_manifest(org.id, manifest.content)
-
-        # Create activation key using default CV and library environment
-        activation_key = entities.ActivationKey(
-            auto_attach=True,
-            content_view=org.default_content_view.id,
-            environment=org.library.id,
-            name=gen_string('alpha'),
-            organization=org,
-        ).create()
-
-        # Walk through the list of subscriptions.
-        # Find the "Red Hat Employee Subscription" and attach it to the
-        # recently-created activation key.
-        for subs in entities.Subscription(organization=org).search():
-            if subs.name == DEFAULT_SUBSCRIPTION_NAME:
-                # 'quantity' must be 1, not subscription['quantity']. Greater
-                # values produce this error: "RuntimeError: Error: Only pools
-                # with multi-entitlement product subscriptions can be added to
-                # the activation key with a quantity greater than one."
-                activation_key.add_subscriptions(data={'quantity': 1, 'subscription_id': subs.id})
-                break
-        cls.org_label = org.label
-        cls.ak_name = activation_key.name
-        cls.org_name = org.name
-
-    def test_positive_connection_option(self):
-        """Verify that '--test-connection' option for redhat-access-insights
-        client rpm tests the connection with the satellite server connection
-        with satellite server
-
-        :id: 167758c9-cbfa-4a81-9a11-27f88aaf9118
-
-        :expectedresults: 'redhat-access-insights --test-connection' should
-            return zero on a successfully registered machine to RHAI service
-        """
-        with VirtualMachine(distro=DISTRO_RHEL6) as vm:
-            vm.configure_rhai_client(self.ak_name, self.org_label, DISTRO_RHEL6)
-            test_connection = vm.run('redhat-access-insights --test-connection')
-            self.logger.info(f'Return code for --test-connection {test_connection.return_code}')
-            self.assertEqual(
-                test_connection.return_code, 0, '--test-connection check was not successful'
-            )
+def test_positive_connection_option(org, activation_key):
+    """Verify that 'insights-client --test-connection' successfully tests the proxy connection via
+    the Satellite.
+
+    :id: 167758c9-cbfa-4a81-9a11-27f88aaf9118
+
+    :expectedresults: 'insights-client --test-connection' should return 0 on a client that is
+    successfully registered to RHAI through Satellite.
+    """
+    with VirtualMachine(distro=DISTRO_DEFAULT) as vm:
+        vm.configure_rhai_client(activation_key.name, org.label, DISTRO_DEFAULT)
+        test_connection = vm.run('insights-client --test-connection')
+        logger.info(f'insights-client --test-connection returned: {test_connection.return_code}')
+        assert test_connection.return_code == 0, 'insights-client --test-connection failed'
```

Comment:
it could be useful to also include stderr

Reply:
I've updated the `assert` to print `return_code`, `stdout`, and `stderr`. I've removed the use of `logger`, since the `robottelo.ssh` module already logs all of this to `robottelo.log` when the command is run.